### PR TITLE
Add reset password

### PR DESCRIPTION
**Pull request by MrTheGood**
...I have no idea what I'm doing

I hope these basics at least make sense
Let me know what should change

### CODE REVIEW

**Comment by MrTheGood** [comment]
@CJVisser kom maar op met die burn

File: Project-Starlord/Controllers/AccountController.cs [hunk 1]
```diff
@@ -161,6 +165,95 @@ public async Task<ActionResult<UserModel>> Register(UserModel userModel)
...
+                model.Token = Convert.ToBase64String(resetTokenBytes);
+
+                //todo: send reset email or whatever
+                // i really don't know what to do here..
+                string message =
+                    "Hi! To reset your password you need to open the following link within 24h: \n\n https://wanderer.app/passwordReset?token=" +
+                    model.Token;
+
+
+                _context.ResetTokens.Add(model.HashToken());
```
Comment (CJVisser) [inline Project-Starlord/Controllers/AccountController.cs:196]: Why do we hash the token?

File: Project-Starlord/Controllers/AccountController.cs [hunk 2]
```diff
@@ -161,6 +165,95 @@ public async Task<ActionResult<UserModel>> Register(UserModel userModel)
...
+
+            ResetTokenModel tokenModel = _context.ResetTokens.First(e => e.Id == id);
+
+            if (DateTime.Now - tokenModel.Expires > TimeSpan.Zero || tokenModel.IsTokenUsed)
+            {
+                throw new UnauthorizedAccessException();
+            }
+
+
+            byte[] hashBytes = Convert.FromBase64String(tokenModel.Token);
```
Comment (CJVisser) [inline Project-Starlord/Controllers/AccountController.cs:223]: If we don't hash we can remove this

File: Project-Starlord/Helpers/ExtensionMethods.cs [hunk 3]
```diff
@@ -35,5 +36,22 @@ public static UserModel WithoutPassword(this UserModel user)
 
             return user;
         }
+
+        public static ResetTokenModel HashToken(this ResetTokenModel resetToken)
```
Comment (CJVisser) [inline Project-Starlord/Helpers/ExtensionMethods.cs:40]: We can remove this

**Review by CJVisser — APPROVED** [review]

File: Project-Starlord/Controllers/AccountController.cs [hunk 4]
```diff
@@ -161,6 +165,95 @@ public async Task<ActionResult<UserModel>> Register(UserModel userModel)
...
+                // Generate reset token
+                byte[] resetTokenBytes;
+                new RNGCryptoServiceProvider().GetBytes(resetTokenBytes = new byte[16]);
+                model.Token = Convert.ToBase64String(resetTokenBytes);
+
+                //todo: send reset email or whatever
+                // i really don't know what to do here..
+                string message =
+                    "Hi! To reset your password you need to open the following link within 24h: \n\n https://wanderer.app/passwordReset?token=" +
+                    model.Token;
```
Comment (MrTheGood) [inline Project-Starlord/Controllers/AccountController.cs:193]: @CJVisser So, how are we going to do this?

Comment (CJVisser) [reply Project-Starlord/Controllers/AccountController.cs:193]: Just let it rest for now. Fill fix later on

**Review by CJVisser — APPROVED** [review]

**Pull request merged**